rickynyairo/StackOverflow-lite
# Ft post and edit answers 159983743

#### What does this PR do?
Creates tests for version 2 of the Stackoverflow-Lite API. The tests were followed up with endpoints to facilitate the tests to pass. 
The user can make requests to the endpoint in order to:

1. Sign Up
2. Log in
3. Get all questions
4. Get a question
5. Post a question
6. Edit a question
7. Post an answer to a question

#### Heroku Link
https://stackoverflow-lite99.herokuapp.com/api/v2

#### Postman 
[![Run in Postman](https://run.pstmn.io/button.svg)](https://app.getpostman.com/run-collection/479da9f494c39acebfd6)

#### Description of Task to be completed
Endpoints have been set up using flask microframework to facilitate communication with the stackoverflow-lite API 

#### How should this be manually tested?
Clone the repository [here](https://github.com/rickynyairo/StackOverflow-lite/)

>git clone https://github.com/rickynyairo/StackOverflow-lite/

Navigate to the API directory
install the requirements (this should be done in a virtual environment)

> pip install -r requirements.txt 

While in the virtual environment (in the API directory), run the following command to perform the tests:

>nosetests app/tests -v

#### Background context
The API defines a set of endpoints for the StackOverflow-Lite project. Version 2 of the API has a database linked to facilitate persistent data structures.

#### What are the relevant pivotal tracker stories?
[#159966795](https://www.pivotaltracker.com/story/show/159966795)
[#159980501](https://www.pivotaltracker.com/story/show/159980501)
[#159842922](https://www.pivotaltracker.com/story/show/159842922)
[#159920351](https://www.pivotaltracker.com/story/show/159920351)
